elyra/cli/pipeline_app.py

```diff
@@ -45,19 +46,39 @@
             ValidationSeverity.Information: 'Information'}
 
 
-def _get_runtime_type(runtime_config: Optional[str]) -> Optional[str]:
+def _get_runtime_config(runtime_config: Optional[str]) -> Optional[dict]:
+    """Fetch runtime configuration"""
     if not runtime_config or runtime_config == 'local':
         # No runtime configuration was specified or it is local.
         # Cannot use metadata manager to determine the runtime type.
-        return 'local'
+        return None
     try:
         metadata_manager = MetadataManager(schemaspace='runtimes')
-        metadata = metadata_manager.get(runtime_config)
-        return metadata.schema_name
+        return metadata_manager.get(runtime_config)
     except Exception as e:
         raise click.ClickException(f'Invalid runtime configuration: {runtime_config}\n {e}')
 
 
+def _get_runtime_type(runtime_config: Optional[str]) -> Optional[str]:
+    """Get runtime type for the provided runtime_config"""
+    rtc = _get_runtime_config(runtime_config)
+    if rtc:
+        return rtc.metadata.get('runtime_type')
+    return None
+
+
+def _get_runtime_schema_name(runtime_config: Optional[str]) -> Optional[str]:
+    """Get runtime schema name for the provided runtime_config"""
+    if not runtime_config or runtime_config == 'local':
+        # No runtime configuration was specified or it is local.
+        # Cannot use metadata manager to determine the runtime type.
+        return 'local'
```

Comment:
I admit it might be a bit confusing but the context in which the result of `_get_runtime_schema_name` and `_get_runtime_config`  is used is different. The return value of this (renamed legacy) method is used by the callers when they invoke other methods (e.g. [here](https://github.com/ptitzler/elyra-2/blob/cli-pipeline-export/elyra/cli/pipeline_app.py#L301-L304)) that expect the value to be `local` instead of `None`. 

If we were to make these two methods return `None`, some of the callers would have to map `None` to `local` to avoid breaking things downstream.

If we were to make these two methods return `'local'` then some other callers (that need to retrieve the actual metadata) would need to have code to understand that `local` means `None` and doesn't hold any metadata. Deep sigh.